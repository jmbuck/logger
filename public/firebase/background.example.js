// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};