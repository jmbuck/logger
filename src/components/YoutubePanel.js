import React, {Component} from 'react'
import '../css/common.css'
import exitIcon from "../img/x.svg"
import DataNav from './DataNav'
import { retrieveFirebaseUserYoutubeVideoData, msToString,
         deleteFirebaseYoutubeData } from "../logger-firebase";
import { auth } from "../database/Auth";

class YoutubePanel extends Component {

	constructor(props) {
		super(props);

		this.state = {
            data: []
		};
		this.handleDelete = this.handleDelete.bind(this);
	}

	retrieve = (user) => {
        if(user) {
            retrieveFirebaseUserYoutubeVideoData(user.uid, (data) => {
                console.log(data)
                this.setState({ data : data})
            })
        }
    };

	componentWillMount() {
	    if(auth.currentUser)
            this.retrieve(auth.currentUser);

	    auth.onAuthStateChanged((user) => {
	        this.retrieve(user);
        })
    }

    handleDelete = (index) => {
        let data = [...this.state.data]
        let channel = data.splice(index, 1)[0].name
        deleteFirebaseYoutubeData(auth.currentUser.uid, channel)
        this.setState({data});
    };

    render() {
        return (
	        <div className="panel">
                <div className="panel-container">
                    <div className="panel-youtube">
                        <DataNav {...this.props}/>
                        <div className="panel-center-content">
	                        <h1>YouTube</h1>
	                        <table border="1px solid black">
                                <thead>
                                <tr>
                                    <th>Channels Watched</th>
                                    <th>Total Time Watched</th>
                                    <th>Delete</th>
                                </tr>
                                </thead>
                                <tbody>
                                {
                                    this.state.data.map((d, index) =>
                                        <tr key={d.id}>
                                            <td>{d.id}</td>
                                            <td>{msToString(d.time)}</td>
                                            <td>
                                                <img src={exitIcon} onClick={() => this.handleDelete(index)} style={{"filter" : "invert(100%)", "width": "25px", "height" : "25px", "cursor" : "pointer"}}/>
                                            </td>
                                        </tr>)
                                }
                                </tbody>
	                        </table>
                        </div>
                    </div>
                </div>
	        </div>
            )
    }
}

export default YoutubePanel