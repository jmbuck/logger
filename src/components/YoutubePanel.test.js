import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import YoutubePanel from "./YoutubePanel";
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("YoutubePanel", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should render component", () => {
        const component = Enzyme.shallow(<YoutubePanel />);

        expect(component).toMatchSnapshot();
    });
});

