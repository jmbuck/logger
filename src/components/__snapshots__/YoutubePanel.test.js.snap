// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubePanel should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YoutubePanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel-container"
>
        <div
                className="panel-youtube"
        >
                <DataNav />
                <div
                        className="panel-center-content"
                >
                        <h1>
                                YouTube
                        </h1>
                        <table
                                border="1px solid black"
                        >
                                <thead>
                                        <tr>
                                                <th>
                                                        Channels Watched
                                                </th>
                                                <th>
                                                        Total Time Watched
                                                </th>
                                                <th>
                                                        Total Videos Watched
                                                </th>
                                                <th>
                                                        Delete
                                                </th>
                                        </tr>
                                </thead>
                                <tbody>
                                        
                                </tbody>
                        </table>
                </div>
        </div>
</div>,
      "className": "panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-youtube"
>
          <DataNav />
          <div
                    className="panel-center-content"
          >
                    <h1>
                              YouTube
                    </h1>
                    <table
                              border="1px solid black"
                    >
                              <thead>
                                        <tr>
                                                  <th>
                                                            Channels Watched
                                                  </th>
                                                  <th>
                                                            Total Time Watched
                                                  </th>
                                                  <th>
                                                            Total Videos Watched
                                                  </th>
                                                  <th>
                                                            Delete
                                                  </th>
                                        </tr>
                              </thead>
                              <tbody>
                                        
                              </tbody>
                    </table>
          </div>
</div>,
        "className": "panel-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DataNav />,
            <div
              className="panel-center-content"
>
              <h1>
                            YouTube
              </h1>
              <table
                            border="1px solid black"
              >
                            <thead>
                                          <tr>
                                                        <th>
                                                                      Channels Watched
                                                        </th>
                                                        <th>
                                                                      Total Time Watched
                                                        </th>
                                                        <th>
                                                                      Total Videos Watched
                                                        </th>
                                                        <th>
                                                                      Delete
                                                        </th>
                                          </tr>
                            </thead>
                            <tbody>
                                          
                            </tbody>
              </table>
</div>,
          ],
          "className": "panel-youtube",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  YouTube
</h1>,
                <table
                  border="1px solid black"
>
                  <thead>
                                    <tr>
                                                      <th>
                                                                        Channels Watched
                                                      </th>
                                                      <th>
                                                                        Total Time Watched
                                                      </th>
                                                      <th>
                                                                        Total Videos Watched
                                                      </th>
                                                      <th>
                                                                        Delete
                                                      </th>
                                    </tr>
                  </thead>
                  <tbody>
                                    
                  </tbody>
</table>,
              ],
              "className": "panel-center-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "YouTube",
                },
                "ref": null,
                "rendered": "YouTube",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "border": "1px solid black",
                  "children": Array [
                    <thead>
                      <tr>
                                            <th>
                                                                  Channels Watched
                                            </th>
                                            <th>
                                                                  Total Time Watched
                                            </th>
                                            <th>
                                                                  Total Videos Watched
                                            </th>
                                            <th>
                                                                  Delete
                                            </th>
                      </tr>
</thead>,
                    <tbody>
                      
</tbody>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <tr>
                        <th>
                                                Channels Watched
                        </th>
                        <th>
                                                Total Time Watched
                        </th>
                        <th>
                                                Total Videos Watched
                        </th>
                        <th>
                                                Delete
                        </th>
</tr>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <th>
                            Channels Watched
</th>,
                          <th>
                            Total Time Watched
</th>,
                          <th>
                            Total Videos Watched
</th>,
                          <th>
                            Delete
</th>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Channels Watched",
                          },
                          "ref": null,
                          "rendered": "Channels Watched",
                          "type": "th",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Total Time Watched",
                          },
                          "ref": null,
                          "rendered": "Total Time Watched",
                          "type": "th",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Total Videos Watched",
                          },
                          "ref": null,
                          "rendered": "Total Videos Watched",
                          "type": "th",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Delete",
                          },
                          "ref": null,
                          "rendered": "Delete",
                          "type": "th",
                        },
                      ],
                      "type": "tr",
                    },
                    "type": "thead",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "tbody",
                  },
                ],
                "type": "table",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-container"
>
          <div
                    className="panel-youtube"
          >
                    <DataNav />
                    <div
                              className="panel-center-content"
                    >
                              <h1>
                                        YouTube
                              </h1>
                              <table
                                        border="1px solid black"
                              >
                                        <thead>
                                                  <tr>
                                                            <th>
                                                                      Channels Watched
                                                            </th>
                                                            <th>
                                                                      Total Time Watched
                                                            </th>
                                                            <th>
                                                                      Total Videos Watched
                                                            </th>
                                                            <th>
                                                                      Delete
                                                            </th>
                                                  </tr>
                                        </thead>
                                        <tbody>
                                                  
                                        </tbody>
                              </table>
                    </div>
          </div>
</div>,
        "className": "panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="panel-youtube"
>
            <DataNav />
            <div
                        className="panel-center-content"
            >
                        <h1>
                                    YouTube
                        </h1>
                        <table
                                    border="1px solid black"
                        >
                                    <thead>
                                                <tr>
                                                            <th>
                                                                        Channels Watched
                                                            </th>
                                                            <th>
                                                                        Total Time Watched
                                                            </th>
                                                            <th>
                                                                        Total Videos Watched
                                                            </th>
                                                            <th>
                                                                        Delete
                                                            </th>
                                                </tr>
                                    </thead>
                                    <tbody>
                                                
                                    </tbody>
                        </table>
            </div>
</div>,
          "className": "panel-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DataNav />,
              <div
                className="panel-center-content"
>
                <h1>
                                YouTube
                </h1>
                <table
                                border="1px solid black"
                >
                                <thead>
                                                <tr>
                                                                <th>
                                                                                Channels Watched
                                                                </th>
                                                                <th>
                                                                                Total Time Watched
                                                                </th>
                                                                <th>
                                                                                Total Videos Watched
                                                                </th>
                                                                <th>
                                                                                Delete
                                                                </th>
                                                </tr>
                                </thead>
                                <tbody>
                                                
                                </tbody>
                </table>
</div>,
            ],
            "className": "panel-youtube",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    YouTube
</h1>,
                  <table
                    border="1px solid black"
>
                    <thead>
                                        <tr>
                                                            <th>
                                                                                Channels Watched
                                                            </th>
                                                            <th>
                                                                                Total Time Watched
                                                            </th>
                                                            <th>
                                                                                Total Videos Watched
                                                            </th>
                                                            <th>
                                                                                Delete
                                                            </th>
                                        </tr>
                    </thead>
                    <tbody>
                                        
                    </tbody>
</table>,
                ],
                "className": "panel-center-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "YouTube",
                  },
                  "ref": null,
                  "rendered": "YouTube",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "border": "1px solid black",
                    "children": Array [
                      <thead>
                        <tr>
                                                <th>
                                                                        Channels Watched
                                                </th>
                                                <th>
                                                                        Total Time Watched
                                                </th>
                                                <th>
                                                                        Total Videos Watched
                                                </th>
                                                <th>
                                                                        Delete
                                                </th>
                        </tr>
</thead>,
                      <tbody>
                        
</tbody>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <tr>
                          <th>
                                                    Channels Watched
                          </th>
                          <th>
                                                    Total Time Watched
                          </th>
                          <th>
                                                    Total Videos Watched
                          </th>
                          <th>
                                                    Delete
                          </th>
</tr>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <th>
                              Channels Watched
</th>,
                            <th>
                              Total Time Watched
</th>,
                            <th>
                              Total Videos Watched
</th>,
                            <th>
                              Delete
</th>,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Channels Watched",
                            },
                            "ref": null,
                            "rendered": "Channels Watched",
                            "type": "th",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Total Time Watched",
                            },
                            "ref": null,
                            "rendered": "Total Time Watched",
                            "type": "th",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Total Videos Watched",
                            },
                            "ref": null,
                            "rendered": "Total Videos Watched",
                            "type": "th",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Delete",
                            },
                            "ref": null,
                            "rendered": "Delete",
                            "type": "th",
                          },
                        ],
                        "type": "tr",
                      },
                      "type": "thead",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [],
                      },
                      "ref": null,
                      "rendered": Array [],
                      "type": "tbody",
                    },
                  ],
                  "type": "table",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
